WeatherStationClient.py

```diff
@@ -0,0 +1,258 @@
+        if sort is not None:
+            uri = uri + f'/{sort}'
+        return await self._send('GET', uri)
+    
+    async def station_transmission_history_between(self, station_id, from_unix_timestamp, to_unix_timestamp, filter_=None, sort=None):
+        """Read transmission history for specific time period. Optionally you can also sort them ASC or DESC and filter."""
+        uri = f'station/{station_id}/history/'
+        if filter_ is not None:
+            uri = uri + f'/{filter_}'
+        uri = uri + f'/from/{from_unix_timestamp}/to/{to_unix_timestamp}'
+        if sort is not None:
+            uri = uri + f'/{sort}'
+        return await self._send('GET', uri)
+    
+    async def station_licenses(self, station_id):
+        """Retrieve all the licenses of your device. They are separated by the service (models, forecast ...)."""
+        return await self._send('GET', f'station/{station_id}/licenses')
+    
+    
+
+class ClientBuilder:
+
+    class _RequestContents:
+        def __init__(self, method, route, body, headers):
+            self.method = method
+            self.route = route
+            self.headers = headers
+            self.body = body
+    
+    class _ConnectionBase:
+        apiURI = 'https://api.fieldclimate.com/v1'
+    
+        async def __aenter__(self):
+            self._session = aiohttp.ClientSession()
+            return ApiClient(self)
+        
+        async def __aexit__(self, exc_type, exc_value, traceback):
+            await self._session.close()
+            
+        @abstractmethod
+        def _modifyRequest(self, requestContents):
+            pass
+        
+        async def _make_request(self, method, route, body=None):
+            requestContents = ClientBuilder._RequestContents(method, route, body, {'Accept': 'application/json'})
+            self._modifyRequest(requestContents)
+            
+            args = [self.apiURI+'/'+requestContents.route]
+            kwargs = {'headers': requestContents.headers}
+            if requestContents.body is not None:
+                kwargs['json'] = body
+            reqMethod = {
+                'GET': self._session.get,
+                'PUT': self._session.put,
+                'DELETE': self._session.delete,
+                'POST': self._session.post
+            }[method]
+            
+            result = await reqMethod(*args, **kwargs)
+            if result.status == 400:
+                raise BadRequestException()
+            if result.status == 401:
+                raise UnauthorizedException()
+            if result.status == 403:
+                raise ForbiddenException()
+            if result.status == 409:
+                raise ConflictException()
+            if result.status == 500:
+                error_message = await result.text()
+                raise ServerErrorException(error_message)
+            
+            if result.status == 200:
+                success = True
+            if result.status == 204:
```

Comment:
Its strange that server return no content with get ie. if object pointed by some keys dont exist it should be error, if query for objects is send end no object match empty array should be returned.

I would prefer returns of response only. In case of 204 None could be returned.

Reply:
I am uncertain if it is true that the server always returns empty response if 204 and non-empty response if 200. Didn't test delete / put methods (unsure if I'm allowed to test them), but from the docs I have feeling that delete user might return 204 AND 200 with no content (but the distinction between 200 and 204 is still meaningful) and also put methods might return 204 with content.

For this reason, in order to be sure not to loose any information, I opted to always provide both status and reponse.